lib/src/react_client/event_helpers.dart

```diff
@@ -758,247 +760,49 @@ SyntheticWheelEvent createSyntheticWheelEvent({
-  /// if the event instance was instantiated as a [SyntheticAnimationEvent]. For this class, this means
-  /// that one of the following fields must be non-null:
-  /// - animationName
-  ///
-  /// The following fields __are not__ considered in type detection:
-  /// - elapsedTime
-  /// - pseudoElement
-  bool get isAnimationEvent {
-    final typedThis = this as SyntheticAnimationEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticPointerEvent].
+  bool get isPointerEvent => hasProperty(this, 'pointerId');
 
-    if (typedThis.animationName != null) return true;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticTouchEvent].
+  bool get isTouchEvent => hasProperty(this, 'targetTouches');
 
-    return false;
-  }
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticTransitionEvent].
+  bool get isTransitionEvent => hasProperty(this, 'propertyName');
 
-  /// Uses Duck Typing to attempt to detect if the event instance is a [SyntheticUIEvent].
-  ///
-  /// __NOTE:__ A field unique to this class must be non-null in order for this to return true, even
-  /// if the event instance was instantiated as a [SyntheticUIEvent]. For this class, this means
-  /// that one of the following fields must be non-null:
-  /// - detail
-  /// - view
-  bool get isUiEvent {
-    final typedThis = this as SyntheticUIEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticAnimationEvent].
+  bool get isAnimationEvent => hasProperty(this, 'animationName');
 
-    if (typedThis.detail != null) return true;
-    if (typedThis.view != null) return true;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticUIEvent].
+  bool get isUiEvent => hasProperty(this, 'detail');
 
-    return false;
-  }
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticWheelEvent].
+  bool get isWheelEvent => hasProperty(this, 'deltaX');
+}
 
-  /// Uses Duck Typing to attempt to detect if the event instance is a [SyntheticWheelEvent].
-  ///
-  /// __NOTE:__ A field unique to this class must be non-null in order for this to return true, even
-  /// if the event instance was instantiated as a [SyntheticWheelEvent]. For this class, this means
-  /// that one of the following fields must be non-null:
-  /// - deltaX
-  /// - deltaMode
-  /// - deltaY
-  /// - deltaZ
-  bool get isWheelEvent {
-    final typedThis = this as SyntheticWheelEvent;
-
-    if (typedThis.deltaX != null) return true;
-    if (typedThis.deltaMode != null) return true;
-    if (typedThis.deltaY != null) return true;
-    if (typedThis.deltaZ != null) return true;
-
-    return false;
-  }
+extension DataTransferHelper on SyntheticMouseEvent {
+  /// Wraps [dataTransfer] in a Dart object, detecting if it is a DOM [DataTransfer] object or a
+  /// JS object that looks like it.
+  SyntheticDataTransfer get wrappedDataTransfer => syntheticDataTransferFactory(dataTransfer);
```

Comment:
Can we make this `dataTransfer` instead of `wrappedDataTransfer` (and remove dataTransfer from the JS class) so that it's backwards-compatible?
```suggestion
  SyntheticDataTransfer get dataTransfer => syntheticDataTransferFactory(getProperty(this, 'dataTransfer'));
```

Otherwise, current usages of `dataTransfer`, expecting it to be a SyntheticDataTransfer, would break.